### ISSUE TITLE
How can I depend on a lower version purs?

### ISSUE THREAD

**Issue by chrissound** [purescript/spago#666]
```
spago build           
[error] Oh noes! It looks like the PureScript version installed on your system is not compatible with the package-set you're using.

installed `purs` version:    0.13.6
minimum package-set version: 0.13.8
```

Getting the very latest of purs installed sounds like a bit of work, if I can instead downgrade the dependency how do I do this? I assume I need to change this line:
```
let upstream =
      https://github.com/purescript/package-sets/releases/download/psc-0.13.8-20200708/packages.dhall sha256:df5b0f1ae92d4401404344f4fb2a7a3089612c9f30066dcddf9eaea4fe780e29
```

Though it's unclear what the sha256 refers to? How do I get this value for an older release like https://github.com/purescript/package-sets/releases/tag/psc-0.13.6-20200507 ?

**Comment by chrissound** [comment]
Hmmm I just spotted the actual `packages.dhall` file at the end but that doesn't seem to work as I expect.

```
wget https://github.com/purescript/package-sets/releases/download/psc-0.13.6-20200507/packages.dhall
sha256sum packages.dhall 
89d54bc428494100ebf4f3afcbe33747b7cee9cde3d76746d3e6a94fd26142f9  packages.dhall
```

However if I add those values to `packages.dhall`:
```
spago build
spago: 
↳ ./spago.dhall
  ↳ ./packages.dhall
    ↳ https://github.com/purescript/package-sets/releases/download/psc-0.13.6-20200507/packages.dhall sha256:89d54bc428494100ebf4f3afcbe33747b7cee9cde3d76746d3e6a94fd26142f9

Error: Import integrity check failed

Expected hash:

↳ 89d54bc428494100ebf4f3afcbe33747b7cee9cde3d76746d3e6a94fd26142f9

Actual hash:

↳ 9c1e8951e721b79de1de551f31ecb5a339e82bbd43300eb5ccfb1bf8cf7bbd62


122│       https://github.com/purescript/package-sets/releases/download/psc-0.13.6-20200507/packages.dhall
123│       sha256:89d54bc428494100ebf4f3afcbe33747b7cee9cde3d76746d3e6a94fd26142f9

/home/chris/NewProjects/PsCharts/packages.dhall:122:7
```

**Comment by chrissound** [comment]
Found a solution here: https://github.com/purescript/spago#override-a-package-in-the-package-set-with-a-remote-one

I omitted the hash and it worked. I'm leaving this open to clarify hash check though.

**Comment by f-f** [comment]
@chrissound I think we have two issues here:
1. the title/main issue is a duplicate of #657 I think?
2. the story around hashes is not super clear - how do you think we could clarify it? What kind of explanation around this would you feel it's appropriate?

**Comment by chrissound** [comment]
1. Hard to say, that seems to be about 'picking' an alternative version. While my question was *how* do I actually do that? 

2. Well, my question here was "how" do I determine the sha256sum? Is it from the tar file from github? Or something else? Cause thats what I tried and yet it generated an incorrect hash iirc.

:) I only looked at this briefly so I might be talking over my head here.

**Comment by f-f** [comment]
Right, great points:
1. there are [many releases](https://github.com/purescript/package-sets/releases) for `package-sets`, so one can just edit the `package.dhall` to point at the desired one
2. the hash is a [Dhall hash](https://docs.dhall-lang.org/discussions/Safety-guarantees.html#code-injection), and Spago will generate one for you so you don't have to worry about it. It's there mostly to ensure that:
  1. the package set is cached
  2. the file it points to will not change over time

Makes sense?

**Comment by f-f** [comment]
I think this has been solved now that we have the `--tag` flag from #680?

See documentation [here](https://github.com/purescript/spago#to-a-specific-release-automatically), your initial question would be solved by using
```
spago upgrade-set --tag psc-0.13.6
```

I wonder how we should document this so that it's discoverable?
